### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by rvashurin**
- Adds autogenerated docs for RTD

### CODE REVIEW

**Comment by IINemo** [comment]
Could you fix the linter problem ?

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+import subprocess
+import sys
+from pathlib import Path
+
+# Assuming conf.py is located in /Users/sergeypetrakov/Documents/Documents/Skoltech_PhD/rtd_to_main/lm-polygraph/src/lm_polygraph/docs
```
Comment (rvashurin) [inline docs/conf.py:13]: This should be portable

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+# Get the absolute path of the root directory
+root_path = Path(__file__).resolve().parents[2]
+
+# Add the src/lm_polygraph directory to the Python path
+sys.path.insert(0, str(root_path / "src" / "lm_polygraph"))
+sys.path.insert(0, str(root_path))
+
+# Install the library using pip
+subprocess.run(["pip", "install", "-e", "."], cwd=root_path)
```
Comment (rvashurin) [inline docs/conf.py:23]: This should probably be version-specific?

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+
+project = "LM-Polygraph"
+copyright = "2023, MBZUAI"
+author = "MBZUAI"
+release = "0.3.0"
```
Comment (rvashurin) [inline docs/conf.py:29]: Is there a way to tie this to actual library version?

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "sphinx.ext.viewcode",
+    "sphinx.ext.autodoc",
+    "sphinx.ext.duration",
+    "sphinx.ext.doctest",
+    "sphinx.ext.autosummary",
+]
+
+templates_path = ["_templates"]
+exclude_patterns = ["Thumbs.db", ".DS_Store"]  # '_build',
+# autodoc_mock_imports = ['lm_polygraph']
```
Comment (rvashurin) [inline docs/conf.py:45]: What are these comments? Do we need them?

File: docs/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+dill>=0.3.5.1
+scipy>=1.9.3
+flask>=2.3.2
+protobuf>=4.23
+einops
+accelerate
+bitsandbytes
+openai
+wget
+
```
Comment (rvashurin) [inline docs/requirements.txt:30]: Do we need a separate requirements.txt here? We already have one in root

File: examples/configs/polygraph_eval_example.yaml [hunk 6]
```diff
@@ -0,0 +1,229 @@
+# Output directory where logs, managers and other evaluation artifacts will be
```
Comment (rvashurin) [inline examples/configs/polygraph_eval_example.yaml:1]: These comments here need to be aligned with the current version of the library.

**Review by rvashurin — COMMENTED** [review]
Some comments

File: docs/conf.py (hunk 1 above)
Comment (SergeyPetrakov) [reply docs/conf.py:13]: this comment was just about the fact that conf.py must be in docs directory

File: docs/conf.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    "sphinx.ext.todo",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.autodoc",
+    "sphinx.ext.duration",
+    "sphinx.ext.doctest",
+    "sphinx.ext.autosummary",
+]
+
+templates_path = ["_templates"]
+exclude_patterns = ["Thumbs.db", ".DS_Store"]  # '_build',
```
Comment (SergeyPetrakov) [inline docs/conf.py:44]: droped

File: docs/conf.py (hunk 4 above)
Comment (SergeyPetrakov) [reply docs/conf.py:45]: droped

File: docs/conf.py (hunk 3 above)
Comment (SergeyPetrakov) [reply docs/conf.py:29]: Theree is a solution using setuptools_scm library

**Review by IINemo — APPROVED** [review]
Looks fine

**Pull request merged**